src/Common/src/System/IO/Win32Marshal.cs

```diff
@@ -71,7 +71,7 @@ internal static Exception GetExceptionForWin32Error(int errorCode, string path)
                     return new IOException(SR.Format(SR.IO_AlreadyExists_Name, path), MakeHRFromErrorCode(errorCode));
 
                 case Interop.Errors.ERROR_FILENAME_EXCED_RANGE:
-                    return new PathTooLongException(SR.IO_PathTooLong);
+                    return new PathTooLongException(SR.Format(SR.IO_PathTooLong, path));
```

Comment:
This should be done same way as it is done in CoreCLR
https://github.com/dotnet/coreclr/blob/6ba74dc2a7194f8d6c86c3aeab572a074ef645c8/src/mscorlib/shared/Interop/Unix/Interop.IOErrors.cs#L140